web/scripts/ui.js

```diff
@@ -517,14 +528,23 @@ export class ComfyUI {
 			this.queue.element,
 			this.history.element,
 			$el("button", {
+				id: "comfy-save-button",
 				textContent: "Save",
 				onclick: () => {
+					let filename = "workflow.json";
+					if (promptFilename.value) {
+						filename = prompt("Save workflow as:", filename);
+						if (filename === null) return;
+						if (!filename.endsWith(".json")) {
```

Comment:
```suggestion
						if (!filename) return;
						if (!filename.toLowerCase().endsWith(".json")) {
```
Also return if filename is an empty string (press OK with nothing entered).

`toLowerCase()` makes that case insenstivie.

Reply:
Yeah thats better thanks.